modules/vaos/app/models/ccra/referral_detail.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Ccra
+  # ReferralDetail represents the detailed information for a single referral from CCRA.
+  class ReferralDetail
+    attr_reader :expiration_date, :type_of_care, :provider_name, :location,
+                :referral_number
+
+    ##
+    # Initializes a new instance of ReferralDetail.
+    #
+    # @param attributes [Hash] A hash containing the referral details from the CCRA response.
+    # @option attributes [Hash] "Referral" The main referral data container.
+    def initialize(attributes)
+      referral = attributes['Referral']
+      return if referral.blank?
+
+      @expiration_date = referral['ReferralExpirationDate']
+      @type_of_care = referral['CategoryOfCare']
+      @provider_name = referral['TreatingProvider']
+      @location = referral['TreatingFacility']
+      @referral_number = referral['ReferralNumber']
```

Comment:
Correct me if I'm wrong, but don't we need to also pass in some kind of data indicating if the referral has been used to book an appointment?

Reply:
I think maybe that could be added in a separate pr for [this ticket](https://github.com/orgs/department-of-veterans-affairs/projects/1323/views/12?pane=issue&itemId=97282835&issue=department-of-veterans-affairs%7Cva.gov-team%7C102813)

Reply:
as per Lee this is data that we will likely be getting from the `VAOS::V2::AppointmentService`, sorry for the confusion